File: 백준/Silver/10828. 스택/스택.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(0);
  stack<int> S;
  int N = 0, value = 0;
  string command = "";
  cin >> N;
  
  for (int i = 0; i < N; i++) {
    cin >> command;
    if(command == "push") {
      cin >> value;
      S.push(value);
    }
    else if(command == "pop") {
      if (S.empty()) cout << -1 << '\n';
      else {
        cout << S.top() << '\n';
        S.pop();
      }
      

    }
    else if(command == "size") {
      cout << S.size() << '\n';
    }
    else if(command == "empty") {
      if (S.empty()) cout << 1 << '\n';
      else cout << 0 << '\n';
    }
    else if(command == "top") {
      if (S.empty()) cout << -1 << '\n';
      else {
        cout << S.top() << '\n';
      }
    }
  }

  return 0;
}

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;
import java.util.Stack;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        String str = br.readLine();
        int comLine = Integer.parseInt(str);
        Stack<Integer> stack = new Stack<>();


        for (int i = 0; i < comLine; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String com = st.nextToken();
            if (com.equals("push")) {
                stack.push(Integer.parseInt(st.nextToken()));
            }
            else if (com.equals("pop")) {
                if (stack.isEmpty()) bw.write("-1"+ "\n");
                else {
                    int popVal = stack.pop();
                    bw.write(popVal + "\n");
                }
            }
            else if (com.equals("size")) {
                bw.write(stack.size() + "\n");
            }
            else if (com.equals("empty")) {
                if (stack.isEmpty()) bw.write("1" + "\n");
                else bw.write("0" + "\n");
            }
            else if (com.equals("top")) {
                if (stack.isEmpty()) bw.write("-1" + "\n");
                else {
                    int topVal = stack.peek();
                    bw.write(topVal + "\n");
                }
            }
        }
        bw.flush();
        bw.close();
    }
}